### PR TITLE
fix(frame): incorrect viewport splitline shadow

### PR DESCRIPTION
**Pull request by lcx-seima**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description

参考 ant-design table 的 [锁列阴影逻辑](https://ant.design/components/table-cn/#components-table-demo-fixed-columns-header)
1. 修复原有视窗左侧阴影错误逻辑
2. 新增视窗右侧阴影线
3. 移除废弃的 showCornerRightShadow 相关代码

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
|  before    |  after |

### 🔍 Self Check before Merge

- [x] Add or update relevant Docs.
- [x] Add or update relevant Demos.
- [x] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/facet/base-facet.ts [hunk 1]
```diff
@@ -1125,8 +1117,9 @@ export abstract class BaseFacet {
         height: cornerHeight,
         viewportWidth: width,
         viewportHeight: height,
-        showCornerRightShadow: !isNil(this.hRowScrollBar),
         // When both a row header and a panel scroll bar exist, show viewport shadow
+        showViewPortLeftShadow:
+          !isNil(this.hRowScrollBar) && !isNil(this.hScrollBar),
```
Comment (xingwanying) [inline packages/s2-core/src/facet/base-facet.ts:1122]: image
如果两个的判断条件是一样的那为什么要用两个变量来控制呢?

Comment (lcx-seima) [reply packages/s2-core/src/facet/base-facet.ts]: cornerRightShadow 剩余逻辑，已移除

File: packages/s2-core/src/common/interface/frame.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export interface FrameConfig {
   height: number;
   viewportWidth: number;
   viewportHeight: number;
-  showCornerRightShadow: boolean;
+  showViewPortLeftShadow: boolean;
```
Comment (wjgogogo) [inline packages/s2-core/src/common/interface/frame.ts:12]:
```suggestion
  showViewportLeftShadow: boolean;
```
`viewport`是一个单词

File: packages/s2-core/src/common/interface/frame.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ export interface FrameConfig {
   height: number;
   viewportWidth: number;
   viewportHeight: number;
-  showCornerRightShadow: boolean;
+  showViewportLeftShadow: boolean;
```
Comment (wjgogogo) [inline packages/s2-core/src/common/interface/frame.ts:12]:
```suggestion
  showViewportaaeftShadow: boolean;
```

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**